crates/ruff_python_formatter/README.md

```diff
@@ -0,0 +1,306 @@
+Django and Zulip. When migrating an existing project from Black to Ruff, you should expect to see
+a few differences on the margins, but the vast majority of your code should be formatted
+identically.
+
+If you identify deviations in your project, spot-check them against the [intentional deviations](#intentional-deviations)
+enumerated below, as well as the [unintentional deviations](https://github.com/astral-sh/ruff/issues?q=is%3Aopen+is%3Aissue+label%3Aformatter)
+filed in the issue tracker. If you've identified a new deviation, feel free to [file an issue](https://github.com/astral-sh/ruff/issues/new).
+
+When run over _non_-Black-formatted code, the formatter makes some different decisions than Black,
+and so more deviations should be expected, especially around the treatment of end-of-line comments.
+For details, see [Black compatibility](#black-compatibility).
+
+## Getting started
+
+The Ruff formatter shipped in an alpha state as part of Ruff v0.0.287.
+
+### CLI
+
+The Ruff formatter is available as a standalone subcommand on the `ruff` CLI:
+
+```console
+❯ ruff format --help
+Run the Ruff formatter on the given files or directories
+
+Usage: ruff format [OPTIONS] [FILES]...
+
+Arguments:
+  [FILES]...  List of files or directories to format
+
+Options:
+      --check
+          Avoid writing any formatted files back; instead, exit with a non-zero status code if any files would have been modified, and zero otherwise
+  -o, --output-file <OUTPUT_FILE>
+          Specify file to write the formatter output to (default: stdout)
+      --target-version <TARGET_VERSION>
+          The minimum Python version that should be supported [possible values: py37, py38, py39, py310, py311, py312]
+      --config <CONFIG>
+          Path to the `pyproject.toml` or `ruff.toml` file to use for configuration
+  -h, --help
+          Print help
+
+File selection:
+      --respect-gitignore  Respect file exclusions via `.gitignore` and other standard ignore files
+      --force-exclude      Enforce exclusions, even for paths passed to Ruff directly on the command-line
+
+Miscellaneous:
+      --isolated                         Ignore all configuration files
+      --stdin-filename <STDIN_FILENAME>  The name of the file when passing it through stdin
+
+Log levels:
+  -v, --verbose  Enable verbose logging
+  -q, --quiet    Print lint violations, but nothing else
+  -s, --silent   Disable all logging (but still exit with status code "1" upon detecting lint violations)
+```
+
+Note: `ruff format` is currently hidden by default and will not be visible when running
+`ruff --help`.
+
+Similar to Black, running `ruff format /path/to/file.py` will format the given file or directory
+in-place, while `ruff format --check /path/to/file.py` will avoid writing any formatted files back,
+instead exiting with a non-zero status code if any files are not already formatted.
+
+In future releases, the Ruff formatter will be integrated into `ruff check`.
```

Comment:
I would not make this promise today because there's an outstanding discussion about whether we want to do this (we're leaning towards yes, but it comes back to how tightly we want to integrate the formatter into ruff)